src/Logging.chpl

```diff
@@ -0,0 +1,88 @@
+     */
+    class Logger {
+        var level = LogLevel.WARN;
+        
+        var warnLevels = new set(LogLevel,[LogLevel.WARN, LogLevel.INFO,
+                           LogLevel.DEBUG]);
+        
+        var criticalLevels =  new set(LogLevel, [LogLevel.CRITICAL, LogLevel.WARN,
+                                         LogLevel.INFO,LogLevel.DEBUG]);
+                                         
+        var errorLevels =  new set(LogLevel,[LogLevel.ERROR, LogLevel.CRITICAL,
+                           LogLevel.WARN, LogLevel.INFO, LogLevel.DEBUG]);
+                             
+        var infoLevels = new set(LogLevel,[LogLevel.INFO,LogLevel.DEBUG]);  
+        
+        var printDate: bool = true;
+        
+        proc debug(moduleName, routineName, lineNumber, msg: string) throws {
+            if level == LogLevel.DEBUG  {
+                writeln(generateLogMessage(moduleName, routineName, lineNumber, 
+                                            msg, "DEBUG"));
+                stdout.flush();
+            }
+        }
+        
+        proc info(moduleName, routineName, lineNumber, msg: string) throws {
+            if infoLevels.contains(level) {
+                writeln(generateLogMessage(moduleName, routineName, lineNumber, 
+                                            msg, "INFO"));
+                stdout.flush();
+            }
+        }
+        
+        proc warn(moduleName, routineName, lineNumber, msg: string) throws {
+            if warnLevels.contains(level) {
+                writeln(generateLogMessage(moduleName, routineName, lineNumber, 
+                                            msg, "WARN"));
+                stdout.flush();
+            }
+        }
+        
+        proc critical(moduleName, routineName, lineNumber, msg: string) throws {
+            writeln(generateLogMessage(moduleName, routineName, lineNumber, 
+                                            msg, "CRITICAL"));
+            stdout.flush();
+        }
+        
+        proc error(moduleName, routineName, lineNumber, msg: string) throws {
+            writeln(generateLogMessage(moduleName, routineName, lineNumber, 
+                                            msg, "ERROR"));
+            stdout.flush();
+        }
+        
+        proc generateLogMessage(moduleName: string, routineName, lineNumber, 
+                           msg, level: string) throws {
+             if printDate {
+                 return "%s [%s] %s Line %i %s [Chapel] %s".format(
+                 generateDateTimeString(), moduleName,routineName,lineNumber, 
+                                     level,msg);
+             } else {
+                 return "[%s] %s Line %i %s [Chapel] %s".format(moduleName, 
+                 routineName,lineNumber,level,msg);            
+             }
+        }
+         
+        proc generateDateTimeString() throws {
+            var dts = datetime.today():string;
```

Comment:
Another minor thing, but it's not free to get the current time. This was historically a system call, though with modern `vDSO` (virtual dynamic shared object) it's all in user-space and only takes ~15 ns (at least based on my last timings for this in https://github.com/chapel-lang/chapel/pull/12244)

I don't think it's worth changing anything here, and the overhead of the getting the time compared to the time to do large array operations should be negligible. I was just looking for sources of overhead while looking at https://github.com/mhmerrill/arkouda/issues/614 so wanted to note it before I forgot.